apps/example/pages/index.tsx

```diff
@@ -1,10 +1,24 @@
-export function Index() {
-  /*
-   * Replace the elements below with your own.
-   *
-   * Note: The corresponding styles are in the ./index.css file.
-   */
-  return <div>Next JS application with frontegg</div>;
+import { GetServerSideProps } from 'next';
+import { AdminPortal, getSession } from '@frontegg/nextjs';
+
+export default function MyPage() {
+  return (
+    <div>
+      <h1>Welcome to NextJS with Frontegg</h1>
+      <button data-test-id="open-admin-portal-btn" onClick={() => AdminPortal.show()}>Open Admin Portal</button>
+    </div>
+  );
 }
 
-export default Index;
+export const getServerSideProps: GetServerSideProps = async (context) => {
+  const session = await getSession(context.req);
+  if (session) {
+    const { data } = await fetch('{external}/product', {
```

Comment:
Is this method necessary with all the details inside? I guess it is an example of how to wrap a page to not be redirected to log in, but can we keep it generic without an actual fetch?